File: usedesk_sdk/src/main/java/ru/usedesk/sdk/ui/knowledgebase/sections/SectionsFragment.java
package ru.usedesk.sdk.ui.knowledgebase.sections;


import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import ru.usedesk.sdk.R;
import ru.usedesk.sdk.domain.entity.knowledgebase.Section;

public class SectionsFragment extends Fragment {

    private RecyclerView recyclerViewSections;
    private TextView textViewLoading;

    public SectionsFragment() {
    }

    public static SectionsFragment getInstance() {
        return new SectionsFragment();
    }


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_sections, container, false);

        textViewLoading = view.findViewById(R.id.tv_loading);
        recyclerViewSections = view.findViewById(R.id.rv_list);

        SectionsViewModel viewModel = ViewModelProviders.of(this)
                .get(SectionsViewModel.class);

        viewModel.getSectionsLiveData()
                .observe(this, this::onSectionsLoaded);

        return view;
    }

    private void onSectionsLoaded(List<Section> sections) {
        recyclerViewSections.setAdapter(new SectionListAdapter(sections));
        textViewLoading.setVisibility(View.GONE);
        recyclerViewSections.setVisibility(View.VISIBLE);
    }
}

File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/pages/sections/SectionsFragment.kt
package ru.usedesk.knowledgebase_gui.pages.sections

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ProgressBar
import androidx.fragment.app.viewModels
import androidx.recyclerview.widget.RecyclerView
import ru.usedesk.common_gui.UsedeskBinding
import ru.usedesk.common_gui.UsedeskFragment
import ru.usedesk.common_gui.inflateItem
import ru.usedesk.common_gui.showInstead
import ru.usedesk.knowledgebase_gui.R

internal class SectionsFragment : UsedeskFragment() {

    private val viewModel: SectionsViewModel by viewModels()
    private lateinit var binding: Binding

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        doInit {
            binding = inflateItem(inflater,
                    container,
                    R.layout.usedesk_fragment_list,
                    R.style.Usedesk_KnowledgeBase) { rootView, defaultStyleId ->
                Binding(rootView, defaultStyleId)
            }

            init()
        }

        return binding.rootView
    }

    private fun init() {
        SectionsAdapter(binding.rvItems,
                viewLifecycleOwner,
                viewModel) {
            getParentListener<IOnSectionClickListener>()?.onSectionClick(it)
        }
        showInstead(binding.pbLoading, binding.rvItems)
        viewModel.sectionsLiveData.observe(viewLifecycleOwner) {
            if (it != null) {
                showInstead(binding.rvItems, binding.pbLoading)
            }
        }
    }

    companion object {
        fun newInstance(): SectionsFragment {
            return SectionsFragment()
        }
    }

    internal class Binding(rootView: View, defaultStyleId: Int) : UsedeskBinding(rootView, defaultStyleId) {
        val rvItems: RecyclerView = rootView.findViewById(R.id.rv_items)
        val pbLoading: ProgressBar = rootView.findViewById(R.id.pb_loading)
    }
}